src/components/TradeAssetSearch/helpers/customAssetSearch.ts

```diff
@@ -0,0 +1,24 @@
+import type { AssetNamespace, ChainId } from '@shapeshiftoss/caip'
+import { ASSET_NAMESPACE, bscChainId, solanaChainId } from '@shapeshiftoss/caip'
+import { PublicKey } from '@solana/web3.js'
+
+export const isValidSolanaAddress = (contractAddress: string) => {
```

Comment:
See below re: `isAddress` naming to align with viem: https://github.com/shapeshift/web/pull/8619/files#r1922032692
Also confirmed that the implementation of `PublicKey` constructor is sane and doesn't do more than we need: https://github.com/shapeshift/web/pull/8619/files#r1922032692